prefect_snowflake/credentials.py

```diff
@@ -119,3 +137,70 @@ def _validate_okta_kwargs(cls, values):
                 "`okta_endpoint` must be provided"
             )
         return values
+
+    @validator("private_key")
+    def _validate_private_key(cls, private_key):
+        """
+        Ensure a private_key looks like a PEM format certificate.
+        """
+        if private_key is None:
+            return None
+
+        assert isinstance(private_key, (SecretBytes, SecretStr))
+
+        pk = private_key.get_secret_value()
+        pk = pk.decode() if isinstance(pk, bytes) else pk
+
+        if not isinstance(pk, str):
+            raise ValueError(_INVALID_PEM_FORMAT_CERTIFICATE)
+
+        if pk.isspace() or len(pk) == 0:
+            return None
+
+        cert_parts = re.match(_SIMPLE_PEM_CERTIFICATE_REGEX, pk)
+        if cert_parts is None:
+            raise ValueError(_INVALID_PEM_FORMAT_CERTIFICATE)
+
+        return private_key
+
+
+def _resolve_pem_certificate(private_key: Union[str, bytes], password: Optional[str]):
+    """
+    Converts a PEM certificate into a DER binary key
+    """
+    # The original key passed from prefect has the last few lines of the cert
+    # concatenated. This query splits the certificate into head+body+footer,
+    # then splits the body on any whitespace, finally the reassemble_cert turns
+    # the cert body back into a certificate that passes validation
+    # in the serialization stage.
+
+    def _rebuild_private_key(private_key: str) -> str:  # pragma: no cover
+        """
+        Disassemble the certificate and rebuild it.
+        """
+        cert_parts = re.match(_SIMPLE_PEM_CERTIFICATE_REGEX, private_key)
+        if cert_parts is None:
+            raise ValueError(_INVALID_PEM_FORMAT_CERTIFICATE)
```

Comment:
This looks the same as lines 160-162. Perhaps can be refactored into its own method?

Reply:
Will do.

Reply:
Much appreciated!